src/Psalm/Internal/Cli/LanguageServer.php

```diff
@@ -276,14 +291,31 @@ function () use ($current_dir, $options, $vendor_dir): ?\Composer\Autoload\Class
             exit;
         }
 
-        $providers = new Providers(
-            new FileProvider,
-            new ParserCacheProvider($config),
-            new FileStorageCacheProvider($config),
-            new ClassLikeStorageCacheProvider($config),
-            new FileReferenceCacheProvider($config),
-            new ProjectCacheProvider(Composer::getLockFilePath($current_dir))
-        );
+        if (isset($options['no-cache']) || isset($options['i'])) {
+            $providers = new Providers(
```

Comment:
Unless something has changed. When I tried this last it totally breaks the LSP. Cache must remain on. A better way to do this is provide the ability to clear cache on boot.

Reply:
I use LSP on some projects that are too small to detect problems with cache. I agree with using cache, but I really want `no-file-cache` as default (`ParserCacheProvider($config, false)`. Every time the didChange event fires, Psalm does a file analysis and caches the files in the project, which actually happens all too often. I hope you will have no problem turning this off.